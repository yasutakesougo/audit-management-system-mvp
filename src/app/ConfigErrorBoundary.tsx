/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';

interface ConfigErrorBoundaryState { hasError: boolean; message?: string }

export class ConfigErrorBoundary extends React.Component<React.PropsWithChildren, ConfigErrorBoundaryState> {
  constructor(props: React.PropsWithChildren) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(err: unknown): ConfigErrorBoundaryState {
    const msg = String((err as any)?.message || err);
    if (/SharePoint 接続設定が未完了/.test(msg) || /VITE_SP_RESOURCE/.test(msg)) {
      return { hasError: true, message: msg };
    }
    return { hasError: true, message: msg };
  }

  componentDidCatch(error: any, info: any) {
    // eslint-disable-next-line no-console
    console.error('[ConfigErrorBoundary]', error, info);
  }

  render() {
    if (!this.state.hasError) return this.props.children;
    return (
      <div style={{ padding: '2rem', fontFamily: 'sans-serif', lineHeight: 1.5 }}>
        <h2 style={{ marginTop: 0 }}>環境設定エラー</h2>
        <pre style={{ whiteSpace: 'pre-wrap', background: '#222', color: '#f88', padding: '1rem', borderRadius: 4, overflowX: 'auto' }}>{this.state.message}</pre>
        <h3>修正手順</h3>
        <ol>
          <li>プロジェクト直下に <code>.env</code> を作成 / 更新</li>
          <li>以下の値を実テナントに合わせて設定:</li>
        </ol>
        <pre style={{ background: '#111', color: '#0f0', padding: '0.75rem' }}>{`VITE_SP_RESOURCE=https://contoso.sharepoint.com\nVITE_SP_SITE_RELATIVE=/sites/AuditSystem`}</pre>
        <ol start={3}>
          <li>プレースホルダ (&lt;yourtenant&gt;, &lt;SiteName&gt;, __FILL_ME__) を全て置換</li>
          <li>開発サーバーを再起動 (Ctrl+C → npm run dev)</li>
        </ol>
        <p style={{ marginTop: '1rem' }}>
          <button
            onClick={() => window.location.reload()}
            style={{
              background: '#1976d2',
              color: '#fff',
              border: 'none',
              padding: '0.5rem 1rem',
              borderRadius: 4,
              cursor: 'pointer',
              marginRight: '0.75rem'
            }}
          >再読み込み</button>
          <a
            href="https://github.com/yasutakesougo/audit-management-system-mvp#environment-variables-env"
            target="_blank" rel="noreferrer"
            style={{ fontSize: '0.875rem' }}
          >README 設定ガイドを開く ↗</a>
        </p>
        <p style={{ marginTop: '0.5rem' }}>解消しない場合は <code>ensureConfig</code> 実装 (src/lib/spClient.ts) を確認してください。</p>
      </div>
    );
  }
}

export default ConfigErrorBoundary;
