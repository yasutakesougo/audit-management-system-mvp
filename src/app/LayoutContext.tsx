import React, { createContext, useContext, useEffect, useMemo, useRef, useState } from 'react';

export type DisplayMode = 'sceneA' | 'hybrid' | 'sceneB';
export type LayoutDensity = 'spacious' | 'standard' | 'compact';

type LayoutState = {
	displayMode: DisplayMode;
	density: LayoutDensity;
	width: number;
	height: number;
	isPortrait: boolean;
	isLandscape: boolean;
};

type LayoutProviderProps = {
	children: React.ReactNode;
	/** 画面幅からの自動判定を上書き（ユーザー設定で使える） */
	forceDensity?: LayoutDensity;
	/** ブレークポイントを差し替えたい場合 */
	breakpoints?: Partial<{ sceneA: number; sceneB: number }>;
};

const defaultBreakpoints = { sceneA: 768, sceneB: 1200 } as const;

const defaultViewport = { width: 1024, height: 768 };
const defaultState: LayoutState = {
	displayMode: 'hybrid',
	density: 'standard',
	...defaultViewport,
	isPortrait: defaultViewport.height >= defaultViewport.width,
	isLandscape: defaultViewport.width > defaultViewport.height,
};

function readViewport() {
	if (typeof window === 'undefined') return defaultViewport;
	return {
		width: window.visualViewport?.width ?? window.innerWidth,
		height: window.visualViewport?.height ?? window.innerHeight,
	};
}

function classifyDisplayMode(width: number, bp: { sceneA: number; sceneB: number }): DisplayMode {
	if (width <= bp.sceneA) return 'sceneA';
	if (width >= bp.sceneB) return 'sceneB';
	return 'hybrid';
}

function resolveDensity(displayMode: DisplayMode): LayoutDensity {
	if (displayMode === 'sceneA') return 'spacious';
	if (displayMode === 'sceneB') return 'compact';
	return 'standard';
}

function computeState(
	width: number,
	height: number,
	bp: { sceneA: number; sceneB: number },
	force?: LayoutDensity,
): LayoutState {
	const displayMode = classifyDisplayMode(width, bp);
	const autoDensity = resolveDensity(displayMode);
	const density = force ?? autoDensity;
	const isPortrait = height >= width;
	const isLandscape = width > height;
	return { displayMode, density, width, height, isPortrait, isLandscape };
}

function safeEquals(a: LayoutState, b: LayoutState) {
	return (
		a.displayMode === b.displayMode &&
		a.density === b.density &&
		a.width === b.width &&
		a.height === b.height &&
		a.isPortrait === b.isPortrait &&
		a.isLandscape === b.isLandscape
	);
}

export const LayoutContext = createContext<LayoutState>(defaultState);

export const LayoutProvider: React.FC<LayoutProviderProps> = ({ children, forceDensity, breakpoints }) => {
	const bp = useMemo(
		() => ({ sceneA: breakpoints?.sceneA ?? defaultBreakpoints.sceneA, sceneB: breakpoints?.sceneB ?? defaultBreakpoints.sceneB }),
		[breakpoints],
	);

	const [state, setState] = useState<LayoutState>(() => {
		const { width, height } = readViewport();
		return computeState(width, height, bp, forceDensity);
	});

	const rafId = useRef<number>(0);

	useEffect(() => {
		if (typeof window === 'undefined') return;

		const update = () => {
			cancelAnimationFrame(rafId.current);
			rafId.current = requestAnimationFrame(() => {
				const { width, height } = readViewport();
				setState(prev => {
					const next = computeState(width, height, bp, forceDensity);
					return safeEquals(prev, next) ? prev : next;
				});
			});
		};

		update();

		const onResize = () => update();
		const onOrientation = () => update();

		window.addEventListener('resize', onResize, { passive: true });
		window.addEventListener('orientationchange', onOrientation, { passive: true });
		const vv = window.visualViewport;
		vv?.addEventListener('resize', onResize, { passive: true });
		vv?.addEventListener('scroll', onResize, { passive: true });

		return () => {
			cancelAnimationFrame(rafId.current);
			window.removeEventListener('resize', onResize);
			window.removeEventListener('orientationchange', onOrientation);
			vv?.removeEventListener('resize', onResize);
			vv?.removeEventListener('scroll', onResize);
		};
	}, [bp.sceneA, bp.sceneB, forceDensity]);

	return <LayoutContext.Provider value={state}>{children}</LayoutContext.Provider>;
};

export function useLayout() {
	return useContext(LayoutContext);
}

export function useDisplayMode() {
	return useLayout().displayMode;
}

export function useDensity() {
	return useLayout().density;
}

export function useViewport() {
	const { width, height } = useLayout();
	return { width, height };
}

export function useOrientation() {
	const { isPortrait, isLandscape } = useLayout();
	return { isPortrait, isLandscape };
}

export const layoutBreakpoints = Object.freeze({ sceneA: defaultBreakpoints.sceneA, sceneB: defaultBreakpoints.sceneB });
