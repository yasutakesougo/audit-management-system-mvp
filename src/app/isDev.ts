import { isDevMode } from '@/lib/env';

export function isDevEnvironment(): boolean {
  return isDevMode();
}
