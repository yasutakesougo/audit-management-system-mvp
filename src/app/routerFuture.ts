export const routerFutureFlags = {
  v7_startTransition: true,
  v7_relativeSplatPath: true,
} as const;
