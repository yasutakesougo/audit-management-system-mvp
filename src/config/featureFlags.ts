import { createContext, createElement, useContext, useMemo, type ReactNode } from 'react';
import { isComplianceFormEnabled, isSchedulesCreateEnabled, isSchedulesFeatureEnabled, type EnvRecord } from '@/lib/env';

export type FeatureFlagSnapshot = {
  schedules: boolean;
  schedulesCreate: boolean;
  complianceForm: boolean;
};

export const resolveFeatureFlags = (envOverride?: EnvRecord): FeatureFlagSnapshot => ({
  schedules: isSchedulesFeatureEnabled(envOverride),
  schedulesCreate: isSchedulesCreateEnabled(envOverride),
  complianceForm: isComplianceFormEnabled(envOverride),
});

const initialSnapshot = resolveFeatureFlags();

let currentSnapshot = initialSnapshot;

export const featureFlags = initialSnapshot;

export const getFeatureFlags = (envOverride?: EnvRecord): FeatureFlagSnapshot => {
  if (envOverride) {
    return resolveFeatureFlags(envOverride);
  }
  return currentSnapshot;
};

export const FeatureFlagsContext = createContext<FeatureFlagSnapshot>(initialSnapshot);

type FeatureFlagsProviderProps = {
  value?: FeatureFlagSnapshot;
  children: ReactNode;
};

export const FeatureFlagsProvider: React.FC<FeatureFlagsProviderProps> = ({ value, children }) => {
  const snapshot = value ?? initialSnapshot;
  const memoized = useMemo(() => {
    currentSnapshot = snapshot;
    return snapshot;
  }, [snapshot]);

  return createElement(FeatureFlagsContext.Provider, { value: memoized }, children);
};

export const useFeatureFlags = (): FeatureFlagSnapshot => useContext(FeatureFlagsContext);

export const useFeatureFlag = (flag: keyof FeatureFlagSnapshot): boolean => {
  const flags = useFeatureFlags();
  return flags[flag];
};
