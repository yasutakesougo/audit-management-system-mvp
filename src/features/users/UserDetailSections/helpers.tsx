import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import type { ReactNode } from 'react';
import type { IUserMaster } from '../types';

const formatDateLabel = (value?: string | null): string => {
  if (!value) return '未設定';
  const [year, month, day] = value.split('T')[0]?.split('-') ?? [];
  if (year && month && day) {
    return `${Number(year)}年${Number(month)}月${Number(day)}日`;
  }
  return value;
};

const resolveUserIdentifier = (user: IUserMaster): string => user.UserID || String(user.Id);

const renderHighlights = (items: string[]): ReactNode => (
  <Box component="ul" sx={{ m: 0, pl: 3, display: 'grid', gap: 1 }}>
    {items.map((item, index) => (
      <Typography key={index} component="li" variant="body2" color="text.secondary">
        {item}
      </Typography>
    ))}
  </Box>
);

export { formatDateLabel, renderHighlights, resolveUserIdentifier };
