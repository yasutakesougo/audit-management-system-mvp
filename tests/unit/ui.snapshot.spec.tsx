import { screen, waitFor } from "@testing-library/react";
import { expect, test, vi } from "vitest";

import AppShell from "@/app/AppShell";
import { routerFutureFlags } from "@/app/routerFuture";
import { FeatureFlagsProvider, featureFlags } from "@/config/featureFlags";
import { renderWithAppProviders } from "../helpers/renderWithAppProviders";

vi.mock("@/lib/spClient", () => ({
  useSP: () => ({
    spFetch: vi.fn(() => Promise.resolve({ ok: true })),
  }),
}));

// Wrap with MemoryRouter since AppShell renders navigation links using router context.
test("AppShell snapshot", async () => {
  const { container } = renderWithAppProviders(
    <FeatureFlagsProvider value={{ ...featureFlags, schedules: true }}>
      <AppShell>
        <div data-testid="snapshot-content">content</div>
      </AppShell>
    </FeatureFlagsProvider>,
    { future: routerFutureFlags }
  );

  await waitFor(() => {
    const statuses = screen.getAllByRole("status");
    expect(statuses.some((status) => status.textContent?.includes("SP Connected"))).toBe(true);
  });

  expect(container).toMatchSnapshot();
});
