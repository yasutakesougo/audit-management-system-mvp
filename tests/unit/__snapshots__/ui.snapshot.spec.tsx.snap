// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppShell snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ptdwpq-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1bb4lqe-MuiToolbar-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h6 css-1bv5yfe-MuiTypography-root"
      >
        磯子区障害者地域活動ホーム
      </div>
      <div
        aria-live="polite"
        class="MuiBox-root css-bq9ckw"
        role="status"
      >
        SP Connected
      </div>
      <button
        aria-label="テーマ切り替え"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="Brightness4Icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6"
          />
        </svg>
      </button>
      <a
        aria-label="監査ログ"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
        href="/audit"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="HistoryIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9m-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8z"
          />
        </svg>
      </a>
      <button
        aria-label="サインイン"
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit css-bnyy0h-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        サインイン
      </button>
    </div>
  </header>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg css-18zj4a3-MuiContainer-root"
    role="main"
  >
    <nav
      aria-label="主要ナビゲーション"
      class="MuiBox-root css-1qm1lh"
      role="navigation"
    >
      <div
        class="MuiStack-root css-ofxwhm-MuiStack-root"
      >
        <a
          aria-current="page"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-1y1eefy-MuiButtonBase-root-MuiButton-root"
          data-testid="nav-dashboard"
          href="/"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-d8i980-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="AssignmentTurnedInRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1M9.29 16.29 6.7 13.7a.996.996 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L10 14.17l5.88-5.88c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41l-6.59 6.59c-.38.39-1.02.39-1.41 0"
              />
            </svg>
          </span>
          記録一覧
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1go7da8-MuiButtonBase-root-MuiButton-root"
          href="/daily"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-d8i980-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="AssignmentTurnedInRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1M9.29 16.29 6.7 13.7a.996.996 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L10 14.17l5.88-5.88c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41l-6.59 6.59c-.38.39-1.02.39-1.41 0"
              />
            </svg>
          </span>
          日次記録
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1go7da8-MuiButtonBase-root-MuiButton-root"
          href="/checklist"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-d8i980-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="ChecklistRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 8c0-.55-.45-1-1-1h-7c-.55 0-1 .45-1 1s.45 1 1 1h7c.55 0 1-.45 1-1m-9 8c0 .55.45 1 1 1h7c.55 0 1-.45 1-1s-.45-1-1-1h-7c-.55 0-1 .45-1 1M10.47 4.63c.39.39.39 1.02 0 1.41l-4.23 4.25c-.39.39-1.02.39-1.42 0L2.7 8.16a.996.996 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0l1.42 1.42 3.54-3.54c.38-.38 1.02-.38 1.4 0m.01 8.01c.39.39.39 1.02 0 1.41L6.25 18.3c-.39.39-1.02.39-1.42 0L2.7 16.16c-.39-.39-.39-1.02 0-1.41s1.02-.39 1.41 0l1.42 1.42 3.54-3.54c.38-.38 1.02-.38 1.41.01"
              />
            </svg>
          </span>
          自己点検
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1go7da8-MuiButtonBase-root-MuiButton-root"
          data-testid="nav-audit"
          href="/audit"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-d8i980-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="AssessmentRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2M8 17c-.55 0-1-.45-1-1v-5c0-.55.45-1 1-1s1 .45 1 1v5c0 .55-.45 1-1 1m4 0c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8c0 .55-.45 1-1 1m4 0c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1s1 .45 1 1v2c0 .55-.45 1-1 1"
              />
            </svg>
          </span>
          監査ログ
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1go7da8-MuiButtonBase-root-MuiButton-root"
          href="/users"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-d8i980-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="PeopleAltRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.67 13.13C18.04 14.06 19 15.32 19 17v3h3c.55 0 1-.45 1-1v-2c0-2.18-3.57-3.47-6.33-3.87"
                fill-rule="evenodd"
              />
              <circle
                cx="9"
                cy="8"
                fill-rule="evenodd"
                r="4"
              />
              <path
                d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4c-.47 0-.91.1-1.33.24C14.5 5.27 15 6.58 15 8s-.5 2.73-1.33 3.76c.42.14.86.24 1.33.24m-6 1c-2.67 0-8 1.34-8 4v2c0 .55.45 1 1 1h14c.55 0 1-.45 1-1v-2c0-2.66-5.33-4-8-4"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          利用者
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1go7da8-MuiButtonBase-root-MuiButton-root"
          href="/staff"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-d8i980-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="BadgeRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 7h-5V4c0-1.1-.9-2-2-2h-2c-1.1 0-2 .9-2 2v3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2M9 12c.83 0 1.5.67 1.5 1.5S9.83 15 9 15s-1.5-.67-1.5-1.5S8.17 12 9 12m3 6H6v-.43c0-.6.36-1.15.92-1.39.64-.28 1.34-.43 2.08-.43s1.44.15 2.08.43c.55.24.92.78.92 1.39zm1-9h-2V4h2zm4.25 7.5h-2.5c-.41 0-.75-.34-.75-.75s.34-.75.75-.75h2.5c.41 0 .75.34.75.75s-.34.75-.75.75m0-3h-2.5c-.41 0-.75-.34-.75-.75s.34-.75.75-.75h2.5c.41 0 .75.34.75.75s-.34.75-.75.75"
              />
            </svg>
          </span>
          職員
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1go7da8-MuiButtonBase-root-MuiButton-root"
          data-testid="nav-admin"
          href="/admin/templates"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-d8i980-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="SettingsRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.5 12c0-.23-.01-.45-.03-.68l1.86-1.41c.4-.3.51-.86.26-1.3l-1.87-3.23c-.25-.44-.79-.62-1.25-.42l-2.15.91c-.37-.26-.76-.49-1.17-.68l-.29-2.31c-.06-.5-.49-.88-.99-.88h-3.73c-.51 0-.94.38-1 .88l-.29 2.31c-.41.19-.8.42-1.17.68l-2.15-.91c-.46-.2-1-.02-1.25.42L2.41 8.62c-.25.44-.14.99.26 1.3l1.86 1.41c-.02.22-.03.44-.03.67s.01.45.03.68l-1.86 1.41c-.4.3-.51.86-.26 1.3l1.87 3.23c.25.44.79.62 1.25.42l2.15-.91c.37.26.76.49 1.17.68l.29 2.31c.06.5.49.88.99.88h3.73c.5 0 .93-.38.99-.88l.29-2.31c.41-.19.8-.42 1.17-.68l2.15.91c.46.2 1 .02 1.25-.42l1.87-3.23c.25-.44.14-.99-.26-1.3l-1.86-1.41c.03-.23.04-.45.04-.68m-7.46 3.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5"
              />
            </svg>
          </span>
          設定管理
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1go7da8-MuiButtonBase-root-MuiButton-root"
          data-testid="nav-schedules"
          href="/schedules/week"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-d8i980-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="EventAvailableRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 10.53c-.29-.29-.77-.29-1.06 0l-4.35 4.35L9 13.29c-.29-.29-.77-.29-1.06 0s-.29.77 0 1.06l1.94 1.94c.39.39 1.02.39 1.41 0l4.7-4.7c.3-.29.3-.77.01-1.06M19 3h-1V2c0-.55-.45-1-1-1s-1 .45-1 1v1H8V2c0-.55-.45-1-1-1s-1 .45-1 1v1H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-1 16H6c-.55 0-1-.45-1-1V8h14v10c0 .55-.45 1-1 1"
              />
            </svg>
          </span>
          スケジュール
        </a>
      </div>
    </nav>
    <div
      data-testid="snapshot-content"
    >
      content
    </div>
  </main>
  <footer
    class="MuiBox-root css-1c1ti6f"
    role="contentinfo"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-puci8u-MuiContainer-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 css-oxfp1h-MuiPaper-root"
        style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiStack-root css-10svwat-MuiStack-root"
        >
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorInfo MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorInfo MuiButton-fullWidth css-1h9i99u-MuiButtonBase-root-MuiButton-root"
            href="/daily/attendance"
            tabindex="0"
          >
            通所管理
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth css-9i5bd4-MuiButtonBase-root-MuiButton-root"
            href="/daily/activity"
            tabindex="0"
          >
            活動日誌入力
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth css-13di261-MuiButtonBase-root-MuiButton-root"
            href="/daily/support"
            tabindex="0"
          >
            支援手順記録入力
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorSecondary MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorSecondary MuiButton-fullWidth css-4xzl9k-MuiButtonBase-root-MuiButton-root"
            href="/daily/health"
            tabindex="0"
          >
            健康記録
          </a>
        </div>
      </div>
    </div>
  </footer>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="fixed bottom-4 right-4 z-50 space-y-2"
    data-testid="toast-announcer"
    role="status"
  />
</div>
`;
